File: typescript/src/nodes/catalogue/errorNode.ts
import { BaseNode, NodeResult } from "@/nodes";
import { IAgentContext } from "@/agent";

class ErrorNode extends BaseNode {
  id = "error";

  async run(ctx: IAgentContext): Promise<NodeResult | void> {
    console.error(ctx.error);
  }
}

export default new ErrorNode();

File: python/nodes/catalogue/error_node.py
from __future__ import annotations

import sys
from typing import Optional

from agentic.agent.core.context import AgentContext
from agentic.nodes.core.node import BaseNode, NodeResult


class ErrorNode(BaseNode):
    id = "error"

    async def run(self, ctx: AgentContext) -> Optional[NodeResult]:
        print(ctx.error, file=sys.stderr)
        return None


error_node = ErrorNode()
